File: examples/java-spi-dev/custom-lazyval-generator/src/test/java/com/acme/lazyval/generator/UtilsGeneratorTest.java
package com.acme.lazyval.generator;

import com.qualityminds.lazyval.testkit.Testkit;
import com.qualityminds.lazyval.testkit.Testresult;
import com.qualityminds.lazyval.testkit.scenarios.Scenario;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;

import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class UtilsGeneratorTest {

    @Test
    public void testGenerator(@TempDir Path tempDir) {
        // GIVEN
        var kit = Testkit.java();
        var scenario = Scenario.Java.Isbn;

        // WHEN
        Testresult.Java result = kit.run(tempDir, scenario);

        // THEN
        var expected = new Testresult.Java.Success("IsbnUtils.java", "Utils.java");
        assertEquals(expected, result);
    }
}

File: examples/kotlin-spi-dev/custom-lazyval-generator/src/test/kotlin/com/acme/lazyval/generator/UtilsGeneratorTest.kt
package com.acme.lazyval.generator

import com.qualityminds.lazyval.testkit.Testkit
import com.qualityminds.lazyval.testkit.Testresult
import com.qualityminds.lazyval.testkit.scenarios.Scenario
import org.junit.jupiter.api.io.TempDir

import java.nio.file.Path
import kotlin.test.Test
import kotlin.test.assertEquals

class UtilsGeneratorTest {

    @Test
    fun testGenerator(@TempDir tempDir: Path) {
        // GIVEN
        val kit = Testkit.kotlin()
        val scenario = Scenario.Kotlin.isbn()

        // WHEN
        val result: Testresult.Kotlin = kit.run(tempDir, scenario)

        // THEN
        val expected = Testresult.Kotlin.Success("Utils.kt")
        assertEquals(expected, result)
    }
}
